lib/src/Uncertainty/Algorithm/Transformation/NatafEllipticalCopula/InverseNatafEllipticalCopulaEvaluation.cxx

```diff
@@ -105,6 +104,32 @@ Point InverseNatafEllipticalCopulaEvaluation::operator () (const Point & inP) co
   return result;
 }
 
+Sample InverseNatafEllipticalCopulaEvaluation::operator () (const Sample & inSample) const
+{
+  const UnsignedInteger dimension = getInputDimension();
+  const UnsignedInteger size = inSample.getSize();
+  // First, correlate the components
+  const Matrix inSampleMatrix(inSample.getDimension(), size, inSample.getImplementation()->getData());
+  const Matrix resultMatrix(cholesky_ * inSampleMatrix);
+  SampleImplementation result(size, dimension);
+  result.setData(*resultMatrix.getImplementation());
+  const Distribution standardMarginal(standardDistribution_.getMarginal(0));
+  // Second, apply the commmon marginal distribution
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+  {
+    const Point resultI(standardMarginal.computeCDF(result.getMarginal(i)).asPoint());
+    for (UnsignedInteger j = 0; j < size; ++j)
+      result(j, i) = resultI[j];
+  }
```

Comment:
@dbarbier Here you can apply standardMarginal.computeCDF() over ALL the values contained in result instead of iterating over the dimensions. Something like:
```
SampleImplementation resultFlatten(size*dimension, 1);
resultFlatten.setData(*resultMatrix.getImplementation());
SampleImplementation result(size, dimension);
result.setData(standardMarginal.computeCDF(resultFlatten).asPoint());
```
It may speed things a little bit if data copy through setData() is faster than SampleImplementation::getMarginal() and pointwise copy.

Reply:
Thanks, updated.